repos/system_upgrade/el7toel8/files/rhel_upgrade.py

```diff
@@ -53,28 +53,45 @@ def configure(self):
                 repo.enable()
 
     def run(self):
-        self.base.add_remote_rpms(self.plugin_data['pkgs_info']['local_rpms'])
 
-        for pkg_spec in self.plugin_data['pkgs_info']['to_remove']:
+        to_install_local = self.plugin_data['pkgs_info']['local_rpms']
+        to_install = self.plugin_data['pkgs_info']['to_install']
+        to_remove = self.plugin_data['pkgs_info']['to_remove']
+        to_upgrade = self.plugin_data['pkgs_info']['to_upgrade']
+
+        # Local (on filesystem) amd remote (with url) packages to be installed
+        # add_remote_rpms() accepts list of packages
+
+        self.base.add_remote_rpms(to_install_local)
+
+        # Packages to be removed
+
+        for pkg_spec in to_remove:
```

Comment:
Open question: 3 processing cycles have much in common. Wondering if refactoring them into a function and reusing it will be readable.
Something like
```
def _process_packages(pkg_set, op):
    for pkg_spec in pkg_set:
        try:
            op(pkg_spec)
        except dnf.exceptions.MarkingError:
            self.pkgs_notfound.append(pkg_spec)
```
And invoking 
```
_process_packages(to_remove, self.base.remove)
_process_packages(to_install, self.base.install)
_process_packages(to_upgrade, self.base.upgrade)
```